cognite/neat/app/api/routers/core.py

```diff
@@ -0,0 +1,51 @@
+import json
+import shutil
+import tempfile
+from pathlib import Path
+from typing import cast
+
+from fastapi import APIRouter, UploadFile
+
+from cognite.neat.rules import importers
+from cognite.neat.rules.models._rules.base import RoleTypes
+
+router = APIRouter()
+
+
+@router.post("/api/core/convert")
+async def convert_data_model_to_rules(file: UploadFile):
+    suffix = Path(cast(str, file.filename)).suffix
+
+    if suffix not in [".xlsx", ".ttl", ".owl", ".json", ".yaml"]:
+        return {
+            "filename": None,
+            "content": None,
+            "issues": [f"File type {suffix} not supported. Supported types are ['.xlsx', '.ttl', '.json', '.yaml']"],
+        }
+
+    with tempfile.NamedTemporaryFile(suffix=suffix, delete=False) as temp:
+        shutil.copyfileobj(file.file, temp)  # type: ignore # known issue with mypy
+        temp_filepath = Path(temp.name)
+
```

Comment:
Looks like we are going to implement a register when we get to the rewrite of the API :)